build/azure-pipelines.yml

```diff
@@ -351,137 +345,160 @@ stages:
-              URL=$(ASPNETCORE_URLS)"
-            displayName: Generate .env
-            workingDirectory: $(Build.SourcesDirectory)/tests/Umbraco.Tests.AcceptanceTest/
+
           - script: npm ci --no-fund --no-audit --prefer-offline
             workingDirectory: $(Build.SourcesDirectory)/tests/Umbraco.Tests.AcceptanceTest/
-            displayName: Run npm ci
-          - pwsh: sqllocaldb start mssqllocaldb
-            displayName: Start localdb (Windows only)
-            condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
-          - pwsh: Invoke-Sqlcmd -Query "CREATE DATABASE $env:UmbracoDatabaseName" -ServerInstance $env:UmbracoDatabaseServer
-            displayName: Create database (Windows only)
-            condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
-          - task: UseDotNet@2
-            displayName: Use .NET $(dotnetVersion)
-            inputs:
-              version: $(dotnetVersion)
-              includePreviewVersions: $(dotnetIncludePreviewVersions)
+            displayName: Restore NPM packages
+
+          # Build application
           - pwsh: |
-              $sha = 'g$(Build.SourceVersion)'.substring(0, 8)
-              docker build -t $(dockerImageName):$sha -f $(dockerfile) .
-              mkdir -p $(Build.ArtifactStagingDirectory)/docker-images
-              docker save -o $(Build.ArtifactStagingDirectory)/docker-images/$(dockerImageName).$sha.tar $(dockerImageName):$sha
-              dotnet dev-certs https -ep ${HOME}/.aspnet/https/aspnetapp.pfx -p $(Umbraco__CMS__Unattended__UnattendedUserPassword)
-              docker run --name $(dockerImageName) -dp 8080:5000 -dp 8443:5001 -e UMBRACO__CMS__GLOBAL__ID=$(UMBRACO__CMS__GLOBAL__ID) -e ASPNETCORE_Kestrel__Certificates__Default__Password="$(Umbraco__CMS__Unattended__UnattendedUserPassword)" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v ${HOME}/.aspnet/https:/https/ $(dockerImageName):$sha
-              docker ps
+              $cmsVersion = "$(Build.BuildNumber)" -replace "\+",".g"
+              dotnet new nugetconfig
+              dotnet nuget add source ./nupkg --name Local
+              dotnet new --install Umbraco.Templates::$cmsVersion
+              dotnet new umbraco --name UmbracoProject --version $cmsVersion --exclude-gitignore --no-restore --no-update-check
+              dotnet restore UmbracoProject
+              dotnet build UmbracoProject --configuration $(buildConfiguration) --no-restore
+              dotnet dev-certs https
+            displayName: Build application
+            workingDirectory: $(Agent.BuildDirectory)/app
+
+          # Start SQL Server LocalDB
+          - pwsh: SqlLocalDB start MSSQLLocalDB
+            displayName: Start SQL Server LocalDB
+            condition: and(succeeded(), eq(variables['CONNECTIONSTRINGS__UMBRACODBDSN_PROVIDERNAME'], 'Microsoft.Data.SqlClient'), eq(variables['Agent.OS'], 'Windows_NT'))
```

Comment:
Once a service container is used for SQL Server, we can potentially remove the usage of this LocalDB instance, since they're using the same database engine under the hood... The install/setup using LocalDB is slightly different though, so that part might still be valuable to test.